### DISCUSSION TITLE
Restrict SQL Lab access to datasource

### DISCUSSION THREAD

**Discussion by victorarbuesmallada** [apache/superset#18369] [Q&A / Help]
Hi all,

We are trying to figure out ways of restricting the datasources we can perform SQL Lab queries and it seems that the most granular thing we can achieve is restrict access per schema, which grants accesss to ALL the datasources under that schema, something that we don't want.

Is there any way we could have a restriction based only on datasource at all?

**Comment by geido** [comment]
Hello @Painyjames I had a look at several combinations of permissions and I could not reach your desired state. You might get better chances of having an answer in the [Superset Slack](https://join.slack.com/t/apache-superset/shared_invite/zt-uxbh5g36-AISUtHbzOXcu0BIj7kgUaw). I'll be happy to get you in touch with someone who might help over there.
